File: seleniumEasyPython/steps/drag_and_drop_steps.py
from behave import Given, When, Then
from selenium.webdriver.common.by import By


@Given(u'the developer is on the drag and drop demo selenium easy website')
def get_website(context):
    context.driver.get("https://demo.seleniumeasy.com/drag-and-drop-demo.html")


@When(u'the developer clicks, drags, and drops the box in the drop zone')
def perform_drag_and_drop(context):
    elements = context.drag_and_drop.draggable_elements()
    context.draggable_count += len(elements)
    for element in elements:
        context.action.drag_and_drop(element, context.drag_and_drop.drop_zone()).perform()


@Then(u'the box gets added to the items list')
def check_items_list(context):
    elements = context.drag_and_drop.drop_zone().find_elements(By.XPATH, '//span[@draggable="true"]')
    assert context.draggable_count == len(elements)

File: seleniumEasyJava/src/test/java/E2E/steps/dragAndDrop/DragAndDropSteps.java
package E2E.steps.dragAndDrop;
import E2E.runner.TestRunner;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class DragAndDropSteps {

        @Given("the developer is on the drag and drop demo selenium easy website")
        public void theDeveloperIsOnTheDragAndDropCheckboxDemoSeleniumEasyWebsite() {
            TestRunner.driver.get("https://demo.seleniumeasy.com/drag-and-drop-demo.html");
        };

        @When("the developer clicks, drags, and drops the box in the drop zone")
        public void theDeveloperClicksDragsAndDropsTheBoxInTheDropZone() {
            TestRunner.action.clickAndHold(TestRunner.dragAndDropPage.draggable1)
                    .moveToElement(TestRunner.dragAndDropPage.dropZone)
                    .release(TestRunner.dragAndDropPage.dropZone)
                    .build();
        }


        @Then("the box gets added to the items list")
        public void theBoxGetsAddedToTheItemsList() {
            WebElement dropped = TestRunner.dragAndDropPage.droppedListDiv.findElement(By.xpath("//span[text()='Draggable 1']"));
            Assert.assertEquals("Draggable 1", dropped.getText());
        };



}
